config/networkconnectivity/config.go

```diff
@@ -0,0 +1,17 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: CC0-1.0
+
+package networkconnectivity
+
+import (
+	"github.com/crossplane/upjet/pkg/config"
+)
+
+// Configure configures individual resources by adding custom
+// ResourceConfigurators.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("google_network_connectivity_service_connection_policy", func(r *config.Resource) {
+		config.MarkAsRequired(r.TerraformResource, "location")
```

Comment:
Why do you add this configuration?

Reply:
Mostly because of the comment [here in the docs](https://github.com/crossplane/upjet/blob/main/docs/configuring-a-resource.md#overriding-terraform-resource-schema)  and because `region` is handled in the same way for example in `google_redis_instance` [here](https://github.com/crossplane-contrib/provider-upjet-gcp/blob/main/config/redis/config.go#L13).

But I realize when checking the `config/schema.json` that `google_redis_instance` contains:
```
              "region": {
                "type": "string",
                "description": "The name of the Redis region of the instance.",
                "description_kind": "plain",
                "optional": true
              },
```

that `google_network_connectivity_service_connection_policy` contains:
```
              "location": {
                "type": "string",
                "description": "The location of the ServiceConnectionPolicy.",
                "description_kind": "plain",
                "required": true
              },
```
So maybe it's not needed? 🤷🏻

Reply:
I quickly looked at the cases where we added the `config.MarkAsRequired(r.TerraformResource, "<parameter>")` configuration. As I see, we added this configuration for the fields/parameters that we use in the external name configuration and that are optional in `config/schema.json`. 
 
We are using the location parameter in the external name (.../locations/{{ .parameters.location }}...) configuration of our resource, but since the location parameter is already required, I think this configuration is not necessary.

Reply:
I'll remove it first thing in the morning.

Reply:
Removed redundant required `location` in b1c12dbf53631a0cacb67250d26a4364c03cdb0b.

Reply:
Thank you 🙏